source/common/http/codec_client.h

```diff
@@ -274,10 +274,21 @@ class CodecClient : protected Logger::Loggable<Logger::Id::client>,
 
 using CodecClientPtr = std::unique_ptr<CodecClient>;
 
+/**
+ * Production implementation that installs a real codec without automatically connecting.
+ */
```

Comment:
I wonder if instead of having two different types of codeclients (one which calls connect and one which doesn't) if we could always go with this new "no connect" codepath. I'm wondering if we would be able to eliminate some complexity by only having one path? Maybe that's not feasible. What do you think?